### PR TITLE
feat: add schema-based contract testing for Firestore data boundaries

### PR DESCRIPTION
**Pull request by snava10**
Implement contract testing (step 4 of zero-slop pipeline) to enforce data consistency between the React Native frontend and Firebase Firestore.

Since Firebase is schemaless and we use SDKs (not REST), traditional Pact contract testing doesn't apply. Instead, we use Zod schemas to validate data at every Firestore read/write boundary at runtime.

What was added:
- component/Core/Contracts.ts: Zod schemas for UserSettings, FirestoreItem, and LocalItem documents with strict enum, type, and field validation
- component/Core/__tests__/Contracts.test.ts: 28 contract tests covering valid/invalid data, enum enforcement, round-trip consistency (local → base64 → Firestore → decode → local), and schema drift detection

What was changed:
- component/Core/Storage.tsx: Added validateUserSettings() and validateFirestoreItem() at Firestore read/write boundaries (getUserSettings, createUserSettings, pullItems) so invalid data throws immediately instead of silently corrupting app state
- component/Core/__tests__/Storage.test.tsx: Fixed 7 existing tests that used invalid mock data (missing timestamps, lowercase enum values) — exactly the kind of drift contract testing is designed to catch
- package.json: Added zod dependency and npm run test:contracts script

### CODE REVIEW

**Comment by snava10** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/snava10/alist/pull/40#pullrequestreview-4062846965)

**Comment by snava10** [comment]
@copilot fix this error

Run npm run test:mutate -- --reporters clear-text,json
> aguslist@2.2.2 test:mutate
> npx stryker run --ignoreStatic --reporters clear-text,json
15:43:16 (2561) WARN ProjectReader Glob pattern "!component/**/*.spec.{ts,tsx}" did not exclude any files.
15:43:16 (2561) WARN ProjectReader Glob pattern "!component/**/index.ts" did not exclude any files.
15:43:16 (2561) INFO ProjectReader Found 15 of 80 file(s) to be mutated.
15:43:17 (2561) INFO Instrumenter Instrumented 15 source file(s) with 744 mutant(s)
15:43:17 (2561) INFO ConcurrencyTokenProvider Creating 2 checker process(es) and 2 test runner process(es).
Error: 15:43:22 (2561) ERROR Stryker Unexpected error occurred while running Stryker StrykerError: Error: An error occurred during initialization of the "typescript" checker. Inner error: Error: Typescript error(s) found in dry run compilation: component/Core/Storage.tsx(175,15): error TS2375: Type '{ value: string; name: string; timestamp: number; userId: string; encrypted?: boolean | undefined; }' is not assignable to type 'AListItem' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.

File: component/Core/Contracts.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * If the app writes data that doesn't match, or Firebase returns unexpected data,
+ * validation will fail and tests will catch it.
+ */
+
+// ── Enums ──
+
+export const BackupCadenceSchema = z.enum(['NONE', 'DAILY', 'INSTANT']);
+
+export const MembershipTypeSchema = z.enum(['FREE', 'PREMIUM']);
+
```
[the remark this answers is not in the document]
Comment (snava10) [reply component/Core/Contracts.ts:16]: @copilot apply changes based on [this feedback](https://github.com/snava10/alist/pull/40#discussion_r3040220627)
